plaso/output/shared_elastic.py

```diff
@@ -170,17 +171,19 @@ def _GetSanitizedEventValues(self, event, event_data, event_tag):
         event_values[attribute_name] = attribute_value
 
     # Add a string representation of the timestamp.
-    try:
-      attribute_value = timelib.Timestamp.RoundToSeconds(event.timestamp)
-    except TypeError as exception:
-      logger.warning((
-          'Unable to round timestamp {0!s}. error: {1!s}. '
-          'Defaulting to 0').format(event.timestamp, exception))
-      attribute_value = 0
-
-    attribute_value = timelib.Timestamp.CopyToIsoFormat(
-        attribute_value, timezone=self._output_mediator.timezone)
-    event_values['datetime'] = attribute_value
+    # TODO: determine why the string representation needs to be rounded in
+    # seconds.
+    number_of_seconds, fraction_of_second = divmod(
+        event.timestamp, definitions.MICROSECONDS_PER_SECOND)
+
+    if fraction_of_second > 500000:
```

Comment:
Since we have a macro for MICROSECONDS_PER_SECOND, should 500000 be replaced by definitions.MICROSECONDS_PER_SECOND/2  .

Reply:
Ack, I'll change it to a class constant `_MICROSECONDS_PER_HALF_SECOND`

Reply:
code was removed instead